openapi/openapi.yml

```diff
@@ -0,0 +1,246 @@
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Room'
+  /room/{roomId}:
+    get:
+      tags:
+        - room
+      summary: Find room by ID
+      operationId: getRoomById
+      parameters:
+        - name: roomId
+          in: path
+          description: ID of room to return
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Room'
+        '400':
+          description: Invalid ID supplied
+        '404':
+          description: Room not found
+    delete:
+      tags:
+        - room
+      summary: Delete a room
+      operationId: deleteRoom
+      parameters:
+        - name: roomId
+          in: path
+          description: Room id to delete
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '400':
+          description: Invalid room value
+  /room/{roomId}/peer:
+    post:
+      tags:
+        - room
+      summary: Add peer
+      operationId: addPeer
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PeerConfig'
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Peer'
+        '500':
+          description: reached max number of peers
+  /room/{roomId}/consumer:
+    post:
+      tags:
+        - room
+      summary: Create consumer
+      operationId: createConsumer
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ConsumerConfig'
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Consumer'
+  /room/{roomId}/producer:
+    post:
+      tags:
+        - room
+      summary: Create producer
+      operationId: createProducer
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ProducerConfig'
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Producer'
+components:
+  schemas:
+    RoomConfig:
+      type: object
+      properties:
+        maxPeers:
+          type: integer
+          format: int64
+          example: 10
```

Comment:
Aren't we missing infiinty here?

Reply:
Probably yes To be honest I wasn't to precies when it comes to some properites types and response status codes :roll_eyes: